desktop/ui/src/main/java/org/datacleaner/windows/JsonDatastoreDialog.java

```diff
@@ -58,20 +58,30 @@ public String getWindowTitle() {
     }
 
     @Override
-    protected JsonDatastore createDatastore(String name, String filename) {
-        final File file = new File(filename);
-        return new JsonDatastore(name, new FileResource(file));
+    protected String getDatastoreIconPath() {
+        return IconUtils.JSON_IMAGEPATH;
     }
 
     @Override
-    protected String getDatastoreIconPath() {
-        return IconUtils.JSON_IMAGEPATH;
+    protected JsonDatastore createDatastore(String name, Resource resource) {
+        return new JsonDatastore(name, resource);
     }
 
     @Override
-    protected void setFileFilters(AbstractResourceTextField<?> filenameField) {
-        filenameField.addChoosableFileFilter(FileFilters.JSON);
-        filenameField.addChoosableFileFilter(FileFilters.ALL);
-        filenameField.setSelectedFileFilter(FileFilters.JSON);
+    protected void initializeFileFilters(ResourceSelector resourceSelector) {
+        resourceSelector.addChoosableFileFilter(FileFilters.JSON);
+        resourceSelector.addChoosableFileFilter(FileFilters.ALL);
+        resourceSelector.setSelectedFileFilter(FileFilters.JSON);
     }
+
+    @Override
+    protected boolean isPreviewTableEnabled() {
+        return true;
+    }
+
+    @Override
+    protected boolean isPreviewDataAvailable() {
+        return showPreview;
```

Comment:
Does this ever change? I guess not. But I cannot remember what the purpose was from the super-class's side. Maybe we need to check something?